### PR TITLE
BAQE-1821: Use mocha-jenkins-reporter only

### PR DESCRIPTION
**Pull request by domhanak**
Update name for test suite of IT tests
Employ mocha-jenkins-reporter and configure it

Checking GHActions behavior....

### CODE REVIEW

**Comment by domhanak** [comment]
Behavior looks good, there is console output and also generated test results

**Comment by domhanak** [comment]
Rebased on master @jstastny-cz @tiagobento combination of mocha-multi-reporters and mocha-jenkins-reporter seems to be the only solution that works well.

*fixed mocha-multi-reporters name

File: packages/vscode-extension-pack-kogito-kie-editors/mocha-reporter-config.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
-  "reporterEnabled": "spec, xunit",
-  "xunitReporterOptions": {
-    "output": "./it-tests/results/junit-vscode-it-tests.xml"
+  "reporterEnabled": "mocha-jenkins-reporter",
+  "mochaJenkinsReporterReporterOptions": {
+    "junit_report_name": "@kogito-tooling/vscode-extension-pack-kogito-kie-editors",
+    "junit_report_path": "./it-tests/results/junit-vscode-it-tests.xml",
```
Comment (tiagobento) [inline packages/vscode-extension-pack-kogito-kie-editors/mocha-reporter-config.json]:
```suggestion
    "junit_report_path": "./dist-it-tests/junit-vscode-it-tests.xml",
```

**Comment by tiagobento** [comment]
Hi @domhanak. Thanks for the PR and for rebasing it. Looks like our new CI is working well :)

If you remember the new "conventions" we're trying to put in place on the kogito-tooling monorepo, artifacts produced by IT tests should go in `dist-it-tests`, so I'd change it as suggested above.

**Comment by domhanak** [comment]
@tiagobento applied the suggestion

**Review by tiagobento — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Pull request merged**